app/goal_routes.py

```diff
@@ -0,0 +1,119 @@
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from flask import Blueprint, jsonify, request
+
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+@goals_bp.route("", methods=["GET"])
+def read_goals():
+    goals = Goal.query.all()
+    goals_response = [{"id": goal.id, "title": goal.title} for goal in goals]
+
+    return jsonify(goals_response), 200
+
+
+@goals_bp.route("", methods=["POST"])
+def add_goal():
+    request_body = request.get_json()
+
+    try:
+        new_goal = Goal(title=request_body["title"])
+
+        db.session.add(new_goal)
+        db.session.commit()
+
+        response = {
+            "goal": {
+                "id": new_goal.id,
+                "title": new_goal.title,
+            }
+        }
+        return jsonify(response), 201
+
+    except KeyError:
+        return jsonify({"details": "Invalid data"}), 400
+
+
+@goals_bp.route("/<id>", methods=["GET"])
+def read_one_goal(id):
+    goal = Goal.query.get(id)
+    if goal is None:
+        return jsonify(None), 404
+
+    return {
+        "goal": {
+            "id": goal.id,
+            "title": goal.title
+        }
+    }
```

Comment:
Instead of re-writing this goal dictionary in every goal route, we could move the logic to a helper method in the goal class that we can invoke like so:

```suggestion
    return {
        "goal": goal.to_dict()
    }
```